.NET/Microsoft.Recognizers.Text.Number/NumberRecognizer.cs

```diff
@@ -1,124 +1,175 @@
-                [typeof(PercentModel)] = new PercentModel(
-                            AgnosticNumberParserFactory.GetParser(AgnosticNumberParserType.Percentage, new GermanNumberParserConfiguration()),
-                            new German.PercentageExtractor())
-            });
+            return GetModel<NumberModel>();
         }
 
-        public IModel GetNumberModel(string culture, bool fallbackToDefaultCulture = true)
+        public OrdinalModel GetOrdinalModel()
         {
-            return GetModel<NumberModel>(culture, fallbackToDefaultCulture, NumberOptions.None.ToString());
+            return GetModel<OrdinalModel>();
         }
 
-        public IModel GetOrdinalModel(string culture, bool fallbackToDefaultCulture = true)
+        public PercentModel GetPercentageModel()
         {
-            return GetModel<OrdinalModel>(culture, fallbackToDefaultCulture, NumberOptions.None.ToString());
+            return GetModel<PercentModel>();
         }
 
-        public IModel GetPercentageModel(string culture, bool fallbackToDefaultCulture = true)
+        public NumberRangeModel GetNumberRangeModel()
         {
-            return GetModel<PercentModel>(culture, fallbackToDefaultCulture, NumberOptions.None.ToString());
+            return GetModel<NumberRangeModel>();
         }
 
-        public IModel GetNumberRangeModel(string culture, bool fallbackToDefaultCulture = true)
+        public static List<ModelResult> RecognizeNumber(string query, string culture, NumberOptions options = NumberOptions.None)
         {
-            return GetModel<NumberRangeModel>(culture, fallbackToDefaultCulture, NumberOptions.None.ToString());
+            return RecognizeByModel(recognizer => recognizer.GetNumberModel(), query, culture, options);
+        }
+
+        public static List<ModelResult> RecognizeOrdinal(string query, string culture, NumberOptions options = NumberOptions.None)
+        {
+            return RecognizeByModel(recognizer => recognizer.GetOrdinalModel(), query, culture, options);
+        }
+
+        public static List<ModelResult> RecognizePercentage(string query, string culture, NumberOptions options = NumberOptions.None)
+        {
+            return RecognizeByModel(recognizer => recognizer.GetPercentageModel(), query, culture, options);
+        }
+
+        public static List<ModelResult> RecognizeNumberRange(string query, string culture, NumberOptions options = NumberOptions.None)
+        {
+            return RecognizeByModel(recognizer => recognizer.GetNumberRangeModel(), query, culture, options);
+        }
+
+        private static List<ModelResult> RecognizeByModel(Func<NumberRecognizer, IModel> getModelFunc,  string query, string culture, NumberOptions options)
+        {
+            var recognizer = new NumberRecognizer(culture, options);
+            var model = getModelFunc(recognizer);
+            return model.Parse(query);
+        }
+
+        protected override void InitializeConfiguration()

```

Comment:
Initialize can't always instantiate all supported models. There are startup and memory consumption costs associated with it. In some scenarios, consumers only want to pay the cost for a particular culture.

Reply:
The `InitializeConfiguration()` method actually registers all supported culture models and stores a `Dictionary` of lamba factories to support lazy model creation.

```C#
RegisterModel<NumberModel>(
    Culture.English,
    (options) => new NumberModel(
        AgnosticNumberParserFactory.GetParser(AgnosticNumberParserType.Number, new EnglishNumberParserConfiguration()),
        English.NumberExtractor.GetInstance(NumberMode.PureNumber)));
```

Reply:
Just a minor clarification first: model _factories_ (i.e. very simple lambas) are registered always. Instances are created on first use controller by the `lazyInitialization` param.
We updated the constructor to initialize the culture specified models and maintain the previous behavior.